vllm/worker/hpu_enc_dec_model_runner.py

```diff
@@ -0,0 +1,604 @@
+import dataclasses
+import gc
+import itertools
+import math
+from array import array
+from functools import partial
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Type, cast
+
+import habana_frameworks.torch as htorch
+import torch
+from vllm_hpu_extension.ops import batch2block, block2batch
+
+from vllm.attention import AttentionMetadata
+from vllm.logger import init_logger
+from vllm.model_executor.layers.sampler import SamplerOutput
+from vllm.sampling_params import SamplingParams
+from vllm.sequence import (IntermediateTensors, SequenceData,
+                           SequenceGroupMetadata)
+from vllm.utils import is_fake_hpu
+from vllm.worker.hpu_model_runner import (HpuModelAdapter, HPUModelRunnerBase,
+                                          ModelInputForHPUWithSamplingMetadata,
+                                          setup_profiler, subtuple)
+from vllm.worker.model_runner_base import (
+    _add_attn_metadata_broadcastable_dict,
+    _add_sampling_metadata_broadcastable_dict)
+
+if TYPE_CHECKING:
+    from vllm.attention.backends.abstract import AttentionBackend
+
+logger = init_logger(__name__)
+
+# These values are assumed to be zero in several places.
+# Use caution when updating them!
+_PAD_SLOT_ID = 0
+_PAD_BLOCK_ID = 0
+
+
+class HpuModelAdapterEncoderDecoder(HpuModelAdapter):
```

Comment:
Should derive from HPUModelRunner instead of HpuModelAdapter

Reply:
I don't understand this. We have two classes, `HPUEncoderDecoderModelRunner` and `HpuModelAdapterEncoderDecoder`.

Reply:
I see, I mis-read the class name, please ignore this comment.